### DISCUSSION TITLE
S3 List Buckets that correspond only to a predefined regex pattern/prefix

### DISCUSSION THREAD

**Discussion by sabouaram** [aws/aws-sdk-go-v2#2701] [General]
Hello dears,
Hope all is good, am contacting to ask if there is an actual list method(s) mainly for buckets based on a defined regex pattern.
Something that helps filtering => like the prefix in the ListObjectsV2Input.
If not, is it possible to add this feature in the SDK ? ( to open a PR )
Thanks !

**Comment by RanVaknin** [comment]
Hi @sabouaram ,

AFAIK there is no way to achieve service side filtering of buckets based on a prefix (or at all). You'll have to do the filtering on the client side after the result from ListObjectsV2.

> If not, is it possible to add this feature in the SDK ? ( to open a PR )

The AWS SDKs are code generated from the API models of each service. For this to be added there has to be 1. server side implementation to support this. 2. API model definition updates that the SDK can get regenerated from. 

In other words, the SDK team cannot "add" this functionality. This has to be implemented on the S3 side.

Is there anything stopping you from filtering on the client side?

Thanks,
Ran~

**Reply by sabouaram** [reply]
@RanVaknin thanks a lot for your response !I was wondering if this feature is feasible on the S3 server side or could be implemented.
Thanks again for your clarification.
Best,
Salim

**Reply by RanVaknin** [reply]
I see that this feature request already exists (`Request #: a2v3800000113iOAAQ`) , but I have no visibility about if / when this would get implemented. 

Since this is not actionable by the SDK team I'm going to go ahead and close this discussion.

If you need anything else please feel free to open a separate one. 

All the best,
Ran~
